pkg/build/oci/oci.go

```diff
@@ -130,21 +134,45 @@ func buildImageFromLayer(layerTarGZ string, ic types.ImageConfiguration, created
 
 	si := signed.Image(v1Image)
 
-	// TODO(#145): Attach the SBOM, e.g.
-	// f, err := static.NewFile(sbom, static.WithLayerMediaType(mt))
-	// if err != nil {
-	// 	return nil, err
-	// }
-	// si, err = ocimutate.AttachFileToImage(si, "sbom", f)
-	// if err != nil {
-	// 	return nil, err
-	// }
+	// Attach the SBOM, e.g.
+	if len(sbomFormats) > 0 {
+		var mt ggcrtypes.MediaType
+		var path string
+		switch sbomFormats[0] {
+		case "spdx":
+			mt = ctypes.SPDXJSONMediaType
+			path = filepath.Join(sbomPath, fmt.Sprintf("sbom-%s.spdx.json", arch.ToAPK()))
+		case "cyclonedx":
+			mt = ctypes.CycloneDXJSONMediaType
+			path = filepath.Join(sbomPath, fmt.Sprintf("sbom-%s.cdx", arch.ToAPK()))
+		default:
+			return nil, fmt.Errorf("unsupported SBOM format: %s", sbomFormats[0])
+		}
+		if len(sbomFormats) > 1 {
+			// When we have multiple formats, warn that we're picking the first.
+			logger.Printf("uploading SBOM with media type: %s", mt)
```

Comment:
nit: I'd change this to actually be a warning, or rephrase the message if you want to keep it as a Printf.